app/directives/proposalList/proposalList.directive.js

```diff
@@ -118,57 +118,53 @@ angular.module('dashboard')
                 }
             }
 
-            function removeProposal(proposal) {
-                $log.debug("dbProposalList: removeProposal: " + proposal);
-                if (angular.isObject(proposal)) {
-                    var search = angular.isArray(self.proposals);
-                    for (var index = self.proposals.length + CONST.NOTFOUND; search && index > CONST.NOTFOUND; index--) {
-                        var prop = self.proposals[index];
-                        if (angular.equals(proposal, prop)) {
-                            self.proposals.splice(index, 1);
-                            search = false;
-                        }
-                    }
-                    checkProposals();
-                    countProposals();
+            function removeProposal(data) {
+                $log.debug("dbProposalList: removeProposal: " + JSON.stringify(data));
+                var value;
+                if (angular.isString(data)) {
+                    value = data;
                 }
-                else {
-                    $log.error('dbProposalList: removeProposal parameter invalid');
+                else if (angular.isObject(data)) {
+                    if (angular.isString(data.proposalGuid)) {
+                        value = data.proposalGuid;
+                    }
+                    else {
+                        value = data;
+                    }
                 }
-            }
 
-            function removeProposalById(guid) {
-                $log.debug("dbProposalList: removeProposalById: " + guid);
-                if (angular.isString(guid)) {
+                if (value) {
                     var search = angular.isArray(self.proposals);
                     for (var index = self.proposals.length + CONST.NOTFOUND; search && index > CONST.NOTFOUND; index--) {
                         var prop = self.proposals[index];
-                        if (angular.equals(guid, prop.proposalGuid)) {
+                        if (angular.equals(value, prop.proposalGuid) || angular.equals(value, prop)) {
```

Comment:
Would be good to comment somewhere in removeProposal() the use-cases when value is string proposalGuid or object